### PR TITLE
Try to update type definition to new version of react-navigation 2.0.1

### PR DESCRIPTION
**Pull request by magrinj**
Hello,

I'm new in type declaration with typescript, so I just try to update the `.d.ts` to the new version of react-navigation, I'm pretty sure it's not enough and I did some mistake, but if that can help... 

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/react-navigation/index.d.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Type definitions for react-navigation 1.5
+// Type definitions for react-navigation 2.0.1
```
Comment (rbuckton) [inline types/react-navigation/index.d.ts]: Version numbers in the header should only include major.minor.

Comment (rbuckton) [reply types/react-navigation/index.d.ts]: When upgrading the major version, please follow the steps outlined here: https://github.com/DefinitelyTyped/DefinitelyTyped#i-want-to-update-a-package-to-a-new-major-version

**Review by rbuckton — DISMISSED** [review]
Please address both my feedback and the build issues.

File: types/react-navigation/index.d.ts (hunk 1 above)
Comment (magrinj) [reply types/react-navigation/index.d.ts]: I made a new push following your documentation, I hope it will be fine :)

**Review by brunolemos — APPROVED** [review]

**Comment by brunolemos** [comment]
Probably because of my tslint config, but I have this after I copied this definitions to my project:

```
ERROR: @types/react-navigation/index.d.ts[522, 5]: Declaration of public instance field not allowed after declaration of public instance method. Instead, this should come at the beginning of the class/interface.
ERROR: @types/react-navigation/index.d.ts[523, 5]: Declaration of public instance field not allowed after declaration of public instance method. Instead, this should come at the beginning of the class/interface.
ERROR: @types/react-navigation/index.d.ts[524, 5]: Declaration of public instance field not allowed after declaration of public instance method. Instead, this should come at the beginning of the class/interface.
ERROR: @types/react-navigation/index.d.ts[525, 5]: Declaration of public instance field not allowed after declaration of public instance method. Instead, this should come at the beginning of the class/interface.
ERROR: @types/react-navigation/index.d.ts[526, 5]: Declaration of public instance field not allowed after declaration of public instance method. Instead, this should come at the beginning of the class/interface.
ERROR: @types/react-navigation/index.d.ts[527, 5]: Declaration of public instance field not allowed after declaration of public instance method. Instead, this should come at the beginning of the class/interface.
ERROR: @types/react-navigation/index.d.ts[531, 5]: Declaration of public instance field not allowed after declaration of public instance method. Instead, this should come at the beginning of the class/interface.
ERROR: @types/react-navigation/index.d.ts[536, 5]: Declaration of public instance field not allowed after declaration of public instance method. Instead, this should come at the beginning of the class/interface.
ERROR: @types/react-navigation/index.d.ts[541, 5]: Declaration of public instance field not allowed after declaration of public instance method. Instead, this should come at the beginning of the class/interface.
ERROR: @types/react-navigation/index.d.ts[542, 5]: Declaration of public instance field not allowed after declaration of public instance method. Instead, this should come at the beginning of the class/interface.
ERROR: @types/react-navigation/index.d.ts[543, 5]: Declaration of public instance field not allowed after declaration of public instance method. Instead, this should come at the beginning of the class/interface.
ERROR: @types/react-navigation/index.d.ts[696, 5]: Declaration of public instance field not allowed after declaration of constructor. Instead, this should come at the beginning of the class/interface.
ERROR: @types/react-navigation/index.d.ts[697, 5]: Declaration of public instance field not allowed after declaration of constructor. Instead, this should come at the beginning of the class/interface.
ERROR: @types/react-navigation/index.d.ts[698, 5]: Declaration of public instance field not allowed after declaration of constructor. Instead, this should come at the beginning of the class/interface.
ERROR: @types/react-navigation/index.d.ts[699, 5]: Declaration of public instance field not allowed after declaration of constructor. Instead, this should come at the beginning of the class/interface.
```

**Comment by magrinj** [comment]
@brunolemos Yes absolutely, it's due to your tsconfig, in `@types/react-navigation` there is actually no tslint rules in the tsconfig file :)
The errors are just ordering errors, let me know if you think it's better to change the order, but comparing to the old version it's not ordering too

**Comment by brunolemos** [comment]
@magrinj Yes that would be great! But if that's not a breaking it's ok if you can't too.

**Comment by magrinj** [comment]
No that's not braking, I don't really have the time to change it actually :/

**Comment by magrinj** [comment]
Any update on this ? @rbuckton 
@DanielRosenwasser

**Comment by DanielRosenwasser** [comment]
Thanks for your patience!

**Comment by magrinj** [comment]
Thanks @DanielRosenwasser! :)

**Comment by magrinj** [comment]
@DanielRosenwasser I think something is wrong because, when I try to install version 2.0.0 from yarn like that `yarn add @types/react-navigation@2.0.0` the version of file in `node_modules/@types/react-navigation/index.d.ts` is 1.5

**Comment by Shidil** [comment]
@magrinj  It seems version 2.0.0 was published with the file of 1.5

**Comment by osdiab** [comment]
Yeah I tried to use it but found that `createStackNavigator` doesn't exist 🤔

**Comment by ifiokjr** [comment]
Any idea of when a fix will be available?

**Comment by magrinj** [comment]
@Shidil Yes apparently :/

@brunolemos @rbuckton @DanielRosenwasser: Someone can check and change that please, thanks in advance! :)
Let me know if I did something wrong

**Comment by magrinj** [comment]
@AdrianoRuberto StackActions was already integrated in version 1.5 ? 
I add anything I can from 2.0 based on the flow file on the official repository, maybe some types are missed

**Comment by Shidil** [comment]
@magrinj 

I don't think something is missing, as I can use react-navigation@2.0.4 by using this file directly

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react-navigation/v2/index.d.ts

**Comment by ifiokjr** [comment]
@Shidil those types are broken as they are missing StackActions. Is anyone addressing this as currently, my team requires the `StackActions.reset` action and can't upgrade to the latest version as a result. 

I'm happy to create a PR.

**Comment by richeterre** [comment]
I also noticed that installing version `2.0.x` of these typings still results in `1.x` types being shipped. Why is there a  `v2` subdirectory at all – shouldn't the versioning be handled by the npm version itself? So if I want `1.x` typings, I'd install a `1.x` version of `@types/react-navigation`.

**Comment by magrinj** [comment]
@dawnmist I take a look again at: https://github.com/DefinitelyTyped/DefinitelyTyped#i-want-to-update-a-package-to-a-new-major-version
My mistake I misunderstood the documentation, you're right, I will change my PR, sorry for the inconvenient

**Comment by richeterre** [comment]
No worries, mistakes happen! Kudos to you for doing all the work 🙇

**Comment by dawnmist** [comment]
@magrinj No worries - I very much appreciate the work you did. :) I was just trying to work out what had happened to result in the 1.5 types being released in the 2.0 package, and that looked like the most likely cause.

I've grabbed a copy of the 2.0 types locally, and I am using [patch-package](https://github.com/ds300/patch-package) to replace the currently-released version with the 2.0 ones you've done for the moment. I find patch-package very useful for applying small patches to packages while waiting for a fixed version to be released. :+1:

**Comment by magrinj** [comment]
@dawnmist @richeterre Thank you for your understanding! :)
I just sent a new pull request with the correction [#26228](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/26228), let me know if you saw something wrong in it.
I used by the past `patch-package` it's a really good package!

**Comment by magrinj** [comment]
@AdrianoRuberto @ifiokjr I don't really have the time to add `StackActions` for the moment, I suggest you to make a PR :)

**Pull request merged**